drivers/watchdog/wdt_wwdg_stm32.c

```diff
@@ -0,0 +1,272 @@
+#elif defined(CONFIG_SOC_SERIES_STM32L0X)
+		LL_APB2_GRP1_EnableClock(LL_APB2_GRP1_PERIPH_DBGMCU);
+#endif
+		LL_DBGMCU_APB1_GRP1_FreezePeriph(LL_DBGMCU_APB1_GRP1_WWDG_STOP);
+	}
+
+	if (options & WDT_OPT_PAUSE_IN_SLEEP) {
+		return -ENOTSUP;
+	}
+
+	/* Enable the WWDG */
+	LL_WWDG_Enable(wwdg);
+
+	return 0;
+}
+
+static int wwdg_stm32_disable(struct device *dev)
+{
+	/* watchdog cannot be stopped once started unless SOC gets a reset */
+	ARG_UNUSED(dev);
+
+	return -EPERM;
+}
+
+static int wwdg_stm32_install_timeout(struct device *dev,
+				      const struct wdt_timeout_cfg *config)
+{
+	struct wwdg_stm32_data *data = WWDG_STM32_DATA(dev);
+	WWDG_TypeDef *wwdg = WWDG_STM32_STRUCT(dev);
+	u32_t timeout = config->window.max * USEC_PER_MSEC;
+	u32_t calculated_timeout;
+	u32_t prescaler = 0U;
+	u32_t counter = 0U;
+
+	if (config->callback != NULL) {
+		data->callback = config->callback;
+	}
+
+	wwdg_stm32_convert_timeout(dev, timeout, &prescaler, &counter);
+
+	calculated_timeout = wwdg_stm32_get_timeout(dev, prescaler, counter);
+	if (!(IS_WWDG_PRESCALER(prescaler) && IS_WWDG_COUNTER(counter) &&
+	      IS_WWDG_TIMEOUT(timeout, calculated_timeout))) {
+		/* One of the parameters provided is invalid */
+		return -EINVAL;
+	}
+
+	data->counter = counter;
+
+	/* Configure WWDG */
+	/* Set the programmable prescaler */
+	LL_WWDG_SetPrescaler(wwdg, prescaler);
+
+	/* Set window the same as the counter to be able to feed the WWDG almost
+	 * immediately
+	 */
+	LL_WWDG_SetWindow(wwdg, counter);
+	LL_WWDG_SetCounter(wwdg, counter);
+
+	return 0;
+}
+
+static int wwdg_stm32_feed(struct device *dev, int channel_id)
+{
+	WWDG_TypeDef *wwdg = WWDG_STM32_STRUCT(dev);
+	struct wwdg_stm32_data *data = WWDG_STM32_DATA(dev);
+
+	ARG_UNUSED(channel_id);
+	LL_WWDG_SetCounter(wwdg, data->counter);
+
+	return 0;
+}
+
+void wwdg_stm32_isr(void *arg)
+{
+	struct device *const dev = (struct device *)arg;
+	struct wwdg_stm32_data *data = WWDG_STM32_DATA(dev);
+	WWDG_TypeDef *wwdg = WWDG_STM32_STRUCT(dev);
+
+	if (LL_WWDG_IsEnabledIT_EWKUP(wwdg)) {
+		if (LL_WWDG_IsActiveFlag_EWKUP(wwdg)) {
+			LL_WWDG_ClearFlag_EWKUP(wwdg);
+			data->callback(dev, 0);
+		}
+	}
+}
+
+static const struct wdt_driver_api wwdg_stm32_api = {
+	.setup = wwdg_stm32_setup,
+	.disable = wwdg_stm32_disable,
+	.install_timeout = wwdg_stm32_install_timeout,
+	.feed = wwdg_stm32_feed,
+};
+
+static int wwdg_stm32_init(struct device *dev)
+{
+	struct device *clk = device_get_binding(STM32_CLOCK_CONTROL_NAME);
+	const struct wwdg_stm32_config *cfg = WWDG_STM32_CFG(dev);
+
+	__ASSERT_NO_MSG(clk);
+
+	clock_control_on(clk, (clock_control_subsys_t *) &cfg->pclken);
+
+	wwdg_stm32_irq_config(dev);
+
+	return 0;
+}
+
+static struct wwdg_stm32_data wwdg_stm32_dev_data = {
+	.counter = WWDG_RESET_LIMIT,
+	.callback = NULL
+};
+
+static struct wwdg_stm32_config wwdg_stm32_dev_config = {
+	.pclken = {
+		.enr = DT_INST_0_ST_STM32_WINDOW_WATCHDOG_CLOCK_BITS,
```

Comment:
Please use same set of #define's in whole driver (few lines below, you're using `DT_WWDT_0_IRQ`)